shared/util.js

```diff
@@ -325,6 +327,130 @@ class Util {
 
     return lines.filter(line => lineNumbersToKeep.has(line.lineNumber));
   }
+
+  /**
+   * Upgrades an lhr object in-place to account for changes in the data structure over major versions.
+   * @param {LH.Result} lhr
```

Comment:
Types are tricky on this if we want this at the module level. Not putting our best type foot forward, but correctly typing this is going to be kind of gross and probably annoying to keep updated :/